Move number of worker threads to configuration

Aloren opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Graylog2/gelfclient?pullRequest=35) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Graylog2/gelfclient?pullRequest=35) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Graylog2/gelfclient?pullRequest=35) it.</sub>

Aloren commented:
@joschi thanks for review! I've added simple test for configuration, but I'm not sure how to test correct initialization of `AbstractGelfTransport`? I've also rebased to current master.

joschi commented:
@Aloren I guess it's good enough for now.

Thanks a lot for your contribution!
